src/tcgwars/logic/groovy/TcgStatics.groovy

```diff
@@ -982,6 +983,83 @@ class TcgStatics {
     }
   }
 
+  static holon_pokemon_energy(Object delegate, Integer rainbowEnergyCount) {
+    delegate.globalAbility { Card thisCard->
+      delayed {
+          before PLAY_CARD, {
+            def validPokemon = thisCard.player.pbg.all.findAll {it.cards.energyCount()}
+            def holonAlreadyPlayed = bg.em().retrieveObject("Holon_Pokemon_Energy") == bg.turnCount
+            if (ef.cardToPlay == thisCard) {
+              if (choose([1,2], ["Pokémon", "Energy"], "Play this card as a Pokémon or as an energy?") == 2) {
+                if (!holonAlreadyPlayed) {
+                  if (validPokemon) {
+                    def energyCard
+                    def energyEquivalent = []
+                    def typeImages = []
+                    rainbowEnergyCount.times {
+                      energyEquivalent.add([R, D, F, G, W, L, M, P, Y])
+                      typeImages.add(RAINBOW)
+                    }
+
+                    energyCard = specialEnergy(new CustomCardInfo(thisCard.staticInfo).setCardTypes(ENERGY, SPECIAL_ENERGY), energyEquivalent) {
+                      typeImagesOverride = typeImages
+                      onPlay {}
+                      onRemoveFromPlay {
+                        bg.em().run(new ChangeImplementation(thisCard, energyCard))
+                      }
+                    }
+                    energyCard.player = thisCard.player
+
+                    def pcs = validPokemon.select("Attach to?")
+
+                    // Select an energy before attachment so that thisCard doesn't show up as an option
+                    def returningEnergy = pcs.cards.filterByType(ENERGY).select(count: 1)
+
+                    bg.em().run(new ChangeImplementation(energyCard, thisCard))
+                    def cannotPlayEnergy = bg.em().run(new PlayEnergy(energyCard, pcs))
+                    if (cannotPlayEnergy) {
+                      bg.em().run(new ChangeImplementation(thisCard, energyCard))
+                    } else {
+                      returningEnergy.moveTo(thisCard.player.pbg.hand)
+
+                      bg.em().storeObject("Holon_Pokemon_Energy", bg.turnCount)
```

Comment:
Is this actually used for anything? I thought it was just a hacky way to make Holon's Pokémon to count as turn attachment. This shouldn't be needed if properly using PlayEnergy instead of skipping PlayEnergy and using AttachEnergy.

Reply:
Good point, I've removed the code that uses this storedObject